### PR TITLE
Change behaviour of 'error' level

### PR DESCRIPTION
**Pull request by djkirkham**
Changed the behaviour of the 'error' level for `iris.FUTURE` options so that if a user sets them to `True` a warning is raised rather than an error.

(As an aside, I'm only able to test this behaviour because we currently have an option which is at 'error' level.)

### CODE REVIEW

**Comment by pelson** [comment]
```
======================================================================

ERROR: test_netcdf_promote (iris.tests.unit.test_Future.Test___setattr__)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-2.0a0-py2.7.egg/iris/tests/unit/test_Future.py", line 55, in test_netcdf_promote

    with self.assertGivesWarning(exp_emsg):

AttributeError: 'Test___setattr__' object has no attribute 'assertGivesWarning'
```

**Comment by pelson** [comment]
Getting a:

```
======================================================================
FAIL: test_invalid_netcdf_no_unlimited_attributes (iris.tests.unit.test_Future.Test___setattr__)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-2.0a0-py2.7.egg/iris/tests/unit/test_Future.py", line 65, in test_invalid_netcdf_no_unlimited_attributes
    future.netcdf_no_unlimited = state
AssertionError: AttributeError not raised
----------------------------------------------------------------------
```

File: lib/iris/__init__.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def __repr__(self):
     def __setattr__(self, name, value):
         if name in self.deprecated_options:
             level = self.deprecated_options[name]
-            if level == 'error':
+            if level == 'error' and not value:
```
Comment (dkillick) [inline lib/iris/__init__.py]: I don't understand the purpose of this change. If `__setattr__` has been called, there will **always** be a `value`:

```python
>>> import iris
>>> # Specifically enable deprecated behaviour.
>>> iris.FUTURE.cell_datetime_objects = False  # key=cell_datetime_objects; value=False
DeprecationWarning...
>>>
```

This is why the test is failing, as it is testing precisely this behaviour, is being passed a value and is thus failing as this `if` block evaluates to False.

File: lib/iris/tests/unit/test_Future.py [hunk 2]
```diff
@@ -43,13 +43,17 @@ def test_invalid_attribute(self):
...
-                future.netcdf_promote = state
+        exp_emsg = "'Future' property 'netcdf_promote' is deprecated"
+        with self.assertWarnsRegexp(exp_emsg):
+            future.netcdf_promote = True
+
+    def test_invalid_netcdf_promote(self):
+        future = Future()
+        exp_emsg = "'Future' property 'netcdf_promote' has been deprecated"
+        with self.assertRaisesRegexp(AttributeError, exp_emsg):
+            future.netcdf_promote = False
```
Comment (dkillick) [inline lib/iris/tests/unit/test_Future.py]: These tests are now wrong. The `netcdf_promote` FUTURE has been removed, so if a user tries to set either `True` and `False` for this flag an error should be raised. Please reset these tests to their original behaviour.

File: lib/iris/__init__.py (hunk 1 above)
Comment (djkirkham) [reply lib/iris/__init__.py]: `not False == True`...

**Comment by djkirkham** [comment]
@dkillick Please see https://github.com/SciTools/iris/pull/2838#discussion_r147371426 for reasoning behind this PR.

File: lib/iris/__init__.py (hunk 1 above)
Comment (djkirkham) [reply lib/iris/__init__.py]: I should have said `not True == False` which is the case when we don't want this `if` block to run.

**Comment by dkillick** [comment]
@djkirkham thanks for the context. Problem is, the suggestion you're working to with this disagrees with the plan for Iris v2.0 that @bjlittle and @pelson put together, which we worked to when dealing with the updates to the FUTURE flags...

**Comment by djkirkham** [comment]
Well, maybe it was a bad plan ¯\_(ツ)_/¯

**Comment by dkillick** [comment]
@djkirkham it seems quite reasonable to me :wink:

File: lib/iris/__init__.py (hunk 1 above)
Comment (dkillick) [reply lib/iris/__init__.py]: Of course it is. That always catches me out.

File: lib/iris/tests/unit/test_Future.py [hunk 3]
```diff
@@ -43,22 +43,29 @@ def test_invalid_attribute(self):
...
-    def test_invalid_netcdf_no_unlimited_attributes(self):
+    def test_invalid_netcdf_promote(self):
         future = Future()
-        states = [True, False]
-        exp_emsg =\
-            "deprecated 'netcdf_no_unlimited' behaviour has been removed"
-        for state in states:
-            with self.assertRaisesRegexp(AttributeError, exp_emsg):
-                future.netcdf_no_unlimited = state
+        exp_emsg = "'Future' property 'netcdf_promote' has been deprecated"
```
[the remark this answers is not in the document]
Comment (djkirkham) [reply lib/iris/tests/unit/test_Future.py]: This one's fine, well spotted on the other two though

[the remark this answers is not in the document]
Comment (dkillick) [reply lib/iris/tests/unit/test_Future.py]: Copy-paste mistakes are my favourite mistakes to make :wink:

**Pull request merged**